source/FreeRTOS_IP_Multicast.c

```diff
@@ -0,0 +1,983 @@
+ *
+ * @return eReleaseBuffer This function usually returns eReleaseBuffer as IGMP reports are
+ * scheduled for later. If however the user has implemented IGMP Snooping, the return is
+ * controlled by the eApplicationIgmpFrameReceivedHook function. That function might return
+ * eFrameConsumed if it decided to forward the frame somewhere.
+ */
+    eFrameProcessingResult_t eProcessIGMPPacket( NetworkBufferDescriptor_t * const pxNetworkBuffer )
+    {
+        eFrameProcessingResult_t eReturn = eReleaseBuffer;
+        IGMPPacket_t * pxIGMPPacket;
+
+        if( pxNetworkBuffer->xDataLength >= sizeof( IGMPPacket_t ) )
+        {
+            pxIGMPPacket = ( IGMPPacket_t * ) pxNetworkBuffer->pucEthernetBuffer;
+
+            if( igmpIGMP_MEMBERSHIP_QUERY == pxIGMPPacket->xIGMPHeader.ucMessageType )
+            {
+                prvScheduleMulticastReports( pdFALSE, ( void * ) &( pxIGMPPacket->xIGMPHeader.ulGroupAddress ), pxNetworkBuffer->pxInterface, ( uint16_t ) pxIGMPPacket->xIGMPHeader.ucMaxResponseTime );
+            }
+
+            #if ( ipconfigIGMP_SNOOPING != 0 )
+                extern eFrameProcessingResult_t eApplicationIgmpFrameReceivedHook( NetworkBufferDescriptor_t * pxNetworkBuffer );
+                eReturn = eApplicationIgmpFrameReceivedHook( pxNetworkBuffer );
+            #endif /* ( ipconfigIGMP_SNOOPING != 0 ) */
+        }
+
+        return eReturn;
+    }
+#endif /*( ipconfigUSE_IPv4 == 1 ) */
+
+#if ( ipconfigUSE_IPv6 == 1 )
+    void vProcessMLDPacket( NetworkBufferDescriptor_t * const pxNetworkBuffer )
+    {
+        /* Note: pxNetworkBuffer->xDataLength was already checked to be >= sizeof( MLDv1_Rx_Packet_t ) */
+        const MLDv1_Rx_Packet_t * pxMLD_Packet;
+
+        if( pxNetworkBuffer->xDataLength >= sizeof( MLDv1_Rx_Packet_t ) )
+        {
+            pxMLD_Packet = ( ( const MLDv1_Rx_Packet_t * ) pxNetworkBuffer->pucEthernetBuffer );
+
+            if( pxMLD_Packet->xMLD.ucTypeOfMessage == ipICMP_MULTICAST_LISTENER_QUERY )
+            {
+                prvScheduleMulticastReports( pdTRUE, ( void * ) ( pxMLD_Packet->xMLD.xGroupAddress.ucBytes ), pxNetworkBuffer->pxInterface, ( FreeRTOS_ntohs( pxMLD_Packet->xMLD.usMaxResponseDelay ) / igmpTIMING_PERIOD_MS ) );
+            }
+
+            #if ( ipconfigIGMP_SNOOPING != 0 )
+                extern eFrameProcessingResult_t eApplicationIgmpFrameReceivedHook( NetworkBufferDescriptor_t * pxNetworkBuffer );
+                ( void ) eApplicationIgmpFrameReceivedHook( pxNetworkBuffer );
+            #endif /* ( ipconfigIGMP_SNOOPING != 0 ) */
+        }
+    }
+#endif /* ( ipconfigUSE_IPv4 == 1 ) */
+
+void vRescheduleAllMulticastReports( NetworkInterface_t * pxInterface,
+                                     BaseType_t xMaxCountdown )
+{
+    const ListItem_t * pxIterator;
+    const ListItem_t * xEnd;
+    uint32_t ulRandom;
+    MCastReportData_t * pxMRD;
+
+    xMaxCountdown = min( 1, xMaxCountdown );
```

Comment:
is this supposed to be max? `xMaxCountdown` is an integer so min is meaningless here.


Also note that you've introduced `min` and `max` into the core source files of this project which are famously not portable. You probably don't want to do this.

Reply:
Done. Also found and removed two more.